Compiled from "CWE617_Reachable_Assertion__length_09.java"
public class testcases.CWE617_Reachable_Assertion.CWE617_Reachable_Assertion__length_09 extends testcasesupport.AbstractTestCase {
  static final boolean $assertionsDisabled;

  public testcases.CWE617_Reachable_Assertion.CWE617_Reachable_Assertion__length_09();
    Code:
       0: aload_0
       1: invokespecial #3                  // Method testcasesupport/AbstractTestCase."<init>":()V
       4: return

  public void bad() throws java.lang.Throwable;
    Code:
       0: getstatic     #9                  // Field $assertionsDisabled:Z
       3: ifne          22
       6: ldc           #15                 // String
       8: invokevirtual #17                 // Method java/lang/String.length:()I
      11: ifgt          22
      14: new           #23                 // class java/lang/AssertionError
      17: dup
      18: invokespecial #25                 // Method java/lang/AssertionError."<init>":()V
      21: athrow
      22: return

  private void good1() throws java.lang.Throwable;
    Code:
       0: getstatic     #9                  // Field $assertionsDisabled:Z
       3: ifne          22
       6: ldc           #26                 // String cwe617
       8: invokevirtual #17                 // Method java/lang/String.length:()I
      11: ifgt          22
      14: new           #23                 // class java/lang/AssertionError
      17: dup
      18: invokespecial #25                 // Method java/lang/AssertionError."<init>":()V
      21: athrow
      22: return

  private void good2() throws java.lang.Throwable;
    Code:
       0: getstatic     #9                  // Field $assertionsDisabled:Z
       3: ifne          22
       6: ldc           #26                 // String cwe617
       8: invokevirtual #17                 // Method java/lang/String.length:()I
      11: ifgt          22
      14: new           #23                 // class java/lang/AssertionError
      17: dup
      18: invokespecial #25                 // Method java/lang/AssertionError."<init>":()V
      21: athrow
      22: return

  public void good() throws java.lang.Throwable;
    Code:
       0: aload_0
       1: invokevirtual #28                 // Method good1:()V
       4: aload_0
       5: invokevirtual #31                 // Method good2:()V
       8: return

  public static void main(java.lang.String[]) throws java.lang.ClassNotFoundException, java.lang.InstantiationException, java.lang.IllegalAccessException;
    Code:
       0: aload_0
       1: invokestatic  #34                 // Method mainFromParent:([Ljava/lang/String;)V
       4: return

  static {};
    Code:
       0: ldc           #10                 // class testcases/CWE617_Reachable_Assertion/CWE617_Reachable_Assertion__length_09
       2: invokevirtual #38                 // Method java/lang/Class.desiredAssertionStatus:()Z
       5: ifne          12
       8: iconst_1
       9: goto          13
      12: iconst_0
      13: putstatic     #9                  // Field $assertionsDisabled:Z
      16: return
}
